app/uk/gov/hmrc/customs/emailfrontend/viewmodels/EmailVerifiedOrChangedViewModel.scala

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.customs.emailfrontend.viewmodels
+
+import uk.gov.hmrc.customs.emailfrontend.config.AppConfig
+
+case class EmailVerifiedOrChangedViewModel(
+                                            email: String,
+                                            referrerUrl: Option[String],
+                                            isVerifyJourney: Boolean,
+                                            appConfig: AppConfig
+                                          ) {
+
+  def titleKey: String =
+    if (isVerifyJourney)  { "customs.emailfrontend.email-verified.title-and-heading" }
+    else { "customs.emailfrontend.email-changed.title-and-heading" }
+
+  def panelKey: String =
+    if (isVerifyJourney) { "customs.emailfrontend.email-verified.panel" }
+    else { "customs.emailfrontend.email-changed.panel" }
+
+  def messageKey: Option[String] = {
```

Comment:
Can we do this as a match case rather than an if? Something like this should work

def messageKey: Option[String] = (hasLink, isVerifyJourney) match {
  case (true, true)  => Some("customs.emailfrontend.email-verified.info")
  case (true, false) => None
  case (_, true)     => Some("customs.emailfrontend.email-verified.info")
  case (_, false)    => Some("customs.emailfrontend.email-confirmed.info")
}

Reply:
Yeah that would be more readable.